tests/python/pants_test/reporting/test_reporting_integration.py

```diff
@@ -254,13 +256,45 @@ def test_zipkin_reporter_for_v2_engine(self):
       num_of_traces = len(ZipkinHandler.traces)
       self.assertEqual(num_of_traces, 1)
 
-      trace = ZipkinHandler.traces[-1]
+      trace = next(iter(ZipkinHandler.traces.values()))
       v2_span_name_part = "Scandir"
       self.assertTrue(any(v2_span_name_part in span['name'] for span in trace),
         "There is no span that contains '{}' in it's name. The trace:{}".format(
         v2_span_name_part, trace
         ))
 
+  def test_zipkin_reporter_multi_threads(self):
+    ZipkinHandler = zipkin_handler()
+    with http_server(ZipkinHandler) as port:
+      endpoint = "http://localhost:{}".format(port)
+      command = [
+        '--reporting-zipkin-endpoint={}'.format(endpoint),
+        'compile',
+        'examples/src/scala/org/pantsbuild/example/several_scala_targets::'
+      ]
+
+      pants_run = self.run_pants(command)
+      self.assert_success(pants_run)
+      num_of_traces = len(ZipkinHandler.traces)
+      self.assertEqual(num_of_traces, 1)
+
+      trace = next(iter(ZipkinHandler.traces.values()))
```

Comment:
I'm very curious, why can't this be `ZipkinHandler.traces.values()[0]`?

Reply:
`values()
Return a new view of the dictionary’s values. See the documentation of view objects.`
[https://docs.python.org/3.3/library/stdtypes.html#dict-views](url)